### ISSUE TITLE
Simplify discovery of packages built from a given repository

### ISSUE THREAD

**Issue by tmat** [dotnet/arcade#414]
It would be useful to have a simple REST API that can be called from each repo’s home page to list all the packages the repo has produced, without the need for devs to install any tool and figure out how to find the packages in it.

For example, I can easily see what SourceLink packages have been published to NuGet.org by sending this query: `https://www.nuget.org/packages?q=Microsoft.SourceLink` 

We should have something similar for dotnet-core feed.

**Comment by tmat** [comment]
@maririos

**Comment by dagood** [comment]
I think there are two scenarios described here:

 1. For a specific package you know a repo publishes, e.g. `Microsoft.SouceLink`, get a list of all versions the repo has ever published, like `https://www.nuget.org/packages?q=Microsoft.SourceLink`.
    * Sounds like adding a service to the central blob feed to make the JSON readable.
 2. For a repo, a list of all packages (id and version) published as a result of the latest build. (A badge would also be nice to see the version in the repo readme without clickthrough.)
    * Sounds like a link to some BAR query.

This issue looks more like (1) to me now, with hints at (2). Should both be tracked? (Can discuss offline later in the meeting, just want to lay out my thoughts.)

**Comment by tmat** [comment]
It's mostly [1]. The set of packages built by a repo doesn't change that often and it's easy for the repository to list them explicitly. If the names of all packages start with some prefix, such as Microsoft.SourceLink, it's even easier.

**Comment by jcagme** [comment]
This is the link to the issue tracking the error we get when trying to list the packages in our blob feed: https://github.com/emgarten/Sleet/issues/23

**Comment by jcagme** [comment]
I've actually talked to Justin and looks like there is no built in way on Sleet for `list`ing packages. All the uploaded packages are stored in a json file that we could work with but a service/app has to be created for that.

**Comment by markwilkie** [comment]
@jcagme - sleet does support the gallery yes?  Perhaps we have that turned off for perf?

**Comment by jcagme** [comment]
Sleet provides the files so a "gallery" can be built on top of it. Files which would help search and autocomplete for example but that's pretty much it. 

We have a file in the store that we could parse and store all the packages currently in the feed. There is no need to turn search/autocomplete on (I think we can live without that)

 The whole UI needs to be built on top of these which should not be hard. Even the boilerplate ASP .Net could be a good fit for this.

> Perhaps we have that turned off for perf?

More than perf these files represented an overhead of functionality and storage that we did not need at the time.

**Comment by alexperovich** [comment]
The biggest problem with a gallery on top of a sleet feed is that the search url returns every package in the feed in all cases. So any gallery that wants to support search on sleet needs to filter the package list locally. That gets slower and slower the more packages that are in the feed. The dotnet-core feed is massive currently and would probably not be fast at all.

**Comment by jcagme** [comment]
yeah I was thinking there would be an indexer service which makes the searching for packages faster -> UI talking to an indexed storage thing

**Comment by alexperovich** [comment]
If we are going to write any kind of service to index the packages then we should just make a full nuget server. Managing a blob feed and a separate index server alongside it is probably too much.

I explored creating an angular app and packaging it into a single index.html we could deploy to the feeds alongside index.json that would provide a ui. This was promising but will probably only work if we stop using one feed for everything we do ever.

**Comment by jcagme** [comment]
Now that we have BAR in place with a REST API on top this is something we could easily build and even something that we could add to the darc CLI. Something like `darc get-assets`

FYI @mmitche

**Comment by riarenas** [comment]
darc get-asset already exists, and barviz shows this info. Closing.
